### PR TITLE
[PAY-1454] Redline mobile header to latest figma design

### PR DESCRIPTION
**Pull request by piazzatron**
### Description

Lots of small changes to bring this to spec, found here: 

https://www.figma.com/file/AUCQosYyHmbyYLWIC6GZLi/%F0%9F%92%AC-Direct-Messaging-(DMs)?type=design&node-id=486-49325&t=eVrcG6GwkX5X2IZC-0

New:
Screen Shot 2023-06-12 at 5 26 12 PM

Old:
Screen Shot 2023-06-12 at 5 26 40 PM

### Dragons

Touches a lot of non-DMs code. Touched some reachability conditional rendering logic.

*Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?*

### How Has This Been Tested?

Simulator

*Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.*

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### Feature Flags ###

*Are all new features properly feature flagged? Describe added feature flags.*

### CODE REVIEW

File: packages/mobile/src/screens/profile-screen/ProfileInfo.tsx [hunk 1]
```diff
@@ -32,20 +32,20 @@ const { getProfileUserId } = profilePageSelectors
...
     ...flexRowCentered(),
-    marginRight: spacing(2)
+    marginRight: spacing(2),
+    height: spacing(6),
+    marginBottom: spacing(1)
   },
   username: {
     ...typography.h1,
-    color: palette.neutral,
-    flexShrink: 1
```
Comment (piazzatron) [inline packages/mobile/src/screens/profile-screen/ProfileInfo.tsx:40]: note to self: need to double check this on smaller device sizes

File: packages/mobile/src/screens/profile-screen/ProfileHeader/Bio.tsx [hunk 2]
```diff
@@ -11,7 +11,8 @@ const MAX_BIO_LINES = 2
 
 const useStyles = makeStyles(({ typography, palette, spacing }) => ({
   root: {
-    marginBottom: spacing(2)
+    marginBottom: spacing(2),
+    lineHeight: typography.fontSize.small * 1.5
```
Comment (dharit-tan) [inline packages/mobile/src/screens/profile-screen/ProfileHeader/Bio.tsx:15]: Just double checking cuz we usually do 1.3!

**Review by dharit-tan — APPROVED** [review]

File: packages/mobile/src/screens/profile-screen/ProfileHeader/Bio.tsx (hunk 2 above)
Comment (piazzatron) [reply packages/mobile/src/screens/profile-screen/ProfileHeader/Bio.tsx:15]: Good callout, spec has 1.5! One of the biggest differences between this and spec tbh

**Comment by piazzatron** [comment]
Noticed spacing issues between name + handle, as well as descender cutoff issue

**Comment by piazzatron** [comment]
> Noticed spacing issues between name + handle, as well as descender cutoff issue

Fixed descenders + badges + name/handle margin in this

**Pull request merged**